src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.html

```diff
@@ -0,0 +1,343 @@
+           class="d-flex">
+        <cds-text-label for="domain_settings"
+                        i18n
+                        cdRequiredField="Domain Settings"
+          >Domain Settings
+          <div class="cds-input-group">
+            <input
+              cdsText
+              type="text"
+              placeholder="Domain Settings..."
+              i18n-placeholder
+              id="domain_settings"
+              name="domain_settings"
+              formControlName="domain_settings"
+              [value]="domainSettingsObject?.realm"
+              (click)="editDomainSettingsModal()"
+              [invalid]="
+                !smbForm.controls.domain_settings.valid && smbForm.controls.domain_settings.dirty
+              "
+            />
+            <cds-icon-button kind="ghost"
+                             (click)="editDomainSettingsModal()"
+                             size="md">
+              <svg cdsIcon="edit"
+                   size="32"
+                   class="cds--btn__icon"
+                  icon></svg>
+            </cds-icon-button>
+
+            <cds-icon-button kind="ghost"
+                             (click)="deleteDomainSettingsModal()"
+                             size="md">
+              <svg cdsIcon="trash-can"
+                   size="32"
+                   class="cds--btn__icon"
+                   icon></svg>
+            </cds-icon-button>
+          </div>
+        </cds-text-label>
+      </div>
+      <span
+        class="invalid-feedback"
+        style="font-size: 85%"
+        *ngIf="smbForm.get('domain_settings').hasError('required')"
+        >This field should contain Realm and Join Sources.</span
+      >
+      <div></div>
+    </div>
+
+    <!-- User Group Settings -->
+    <ng-container formArrayName="joinSources"
+                  *ngFor="let dns of joinSources.controls; index as i">
+      <div
+        cdsRow
+        *ngIf="this.smbForm.get('auth_mode').value === 'user'"
+        class="form-item form-item-append"
+      >
+        <div cdsCol
+             [columnNumbers]="{ lg: 8 }">
+          <input
+            cdsText
+            type="text"
+            placeholder="User Group Id"
+            label="User"
+            [formControlName]="i"
+            [invalid]="
+              smbForm.controls['joinSources'].controls[i].invalid &&
+              smbForm.controls['joinSources'].dirty
+            "
+            [invalidText]="refError"
+            modal-primary-focus
+            i18n
+          />
+          <ng-template #refError>
+            <span
+              class="invalid-feedback"
+              *ngIf="smbForm.showError('joinSources[i]', formDir, 'required')"
+              i18n
+              >This field is required.</span
+            >
+          </ng-template>
+        </div>
+        <div cdsCol
+             [columnNumbers]="{ lg: 1 }"
+             class="item-action-btn">
+          <cds-icon-button
+            kind="tertiary"
```

Comment:
Let's unanimously call it danger when its a delete icon
```suggestion
            kind="danger"
```